### PR TITLE
Refactor KME with registration API changes

### PR DESCRIPTION
**Pull request by Ram-srini**
1. Implement Registration API for KME
2. Clean up extended work order info class for KME

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

### CODE REVIEW

File: tc/sgx/trusted_worker_manager/enclave/kme_workload/kme_workload_plug-in.cpp [hunk 1]
```diff
@@ -59,10 +82,91 @@ void KMEWorkloadProcessor::GetUniqueId(
...
+    if (search != this->sig_key_map.end()) {
+        /// Compare MRENCLAVE value
+        EnclaveData* enclaveData = EnclaveData::getInstance();
+        std::string ext_data(enclaveData->get_extended_data());
+        std::string mr_enclave_str = ByteArrayToString(mr_enclave);
+        if (ext_data.compare(mr_enclave_str) != 0) {
+            this->SetStatus((int)ERR_MRENCLAVE_NOT_MATCH,
+                out_work_order_data);
+            ThrowIf<ValueError>(true, "WPE MRENCLAVE value didn't match");
+        }
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave/kme_workload/kme_workload_plug-in.cpp:153]: Need to check if WPE is already registered with KME using this unique_key (verification_key) generated by KME

File: tc/sgx/trusted_worker_manager/enclave/kme_workload/kme_workload_plug-in.cpp [hunk 2]
```diff
@@ -59,10 +82,91 @@ void KMEWorkloadProcessor::GetUniqueId(
...
+        verification_key_hash, e_key);
+    if (err != 0) {
+        this->SetStatus(err, out_work_order_data);
+        ThrowIf<ValueError>(true, "WPE attestation verification failed");
+    }
+    auto search = this->sig_key_map.find(verification_key_hash);
+    if (search != this->sig_key_map.end()) {
+        /// Compare MRENCLAVE value
+        EnclaveData* enclaveData = EnclaveData::getInstance();
+        std::string ext_data(enclaveData->get_extended_data());
```
[the remark this answers is not in the document]
Comment (Ram-srini) [reply tc/sgx/trusted_worker_manager/enclave/kme_workload/kme_workload_plug-in.cpp:147]: Manju is passing through enclave_data other we have to change it in all workloads . @manju956  is my understanding right?

[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/kme_workload/kme_workload_plug-in.cpp:147]: we do persist ext_data in EnclaveData. But you can expect the same parameter as ext_work_order_data to workload as i am already doing it for pre-processing..

**Comment by Ram-srini** [comment]
It is taken care in https://github.com/hyperledger/avalon/pull/512. closing this PR.

**Pull request closed without merging**